examples/rest-react/src/LoginForm.test.js

```diff
@@ -1,22 +1,18 @@
 import React from 'react'
-import { render, fireEvent, screen } from '@testing-library/react'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 import { LoginForm } from './LoginForm'
 
 describe('LoginForm', () => {
   it('should allow a user to log in', async () => {
     render(<LoginForm />)
 
-    fireEvent.change(screen.getByLabelText(/username/i), {
-      target: { value: 'johnUser' },
-    })
-    fireEvent.click(screen.getByText(/submit/i))
+    await userEvent.type(screen.getByLabelText(/username/i), 'johnUser')
 
-    const userId = await screen.findByTestId('userId')
```

Comment:
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-querying-by-text